src/couch_changes.erl

```diff
@@ -349,6 +366,18 @@ check_docids(_) ->
     throw({bad_request, Msg}).
 
 
+check_selector(Selector={_}) ->
+    try
+        mango_selector:normalize(Selector)
+    catch
+        {mango_error, _Mod, Reason} -> % see MANGO_ERROR(R) in mango.hrl
+            Msg = io_lib:format("Selector error: ~p", [Reason]),
```

Comment:
You want `mango_error:info(Mod, Reason) which gives you back a {StatusCode, Error, Reason} triple where Reason is formatted correctly.